engine/router.go

```diff
@@ -336,6 +336,80 @@ func (r *Router) subscribe(rid, uid, cid string) (*webrtc.SessionDescription, er
 	}
 }
 
+func (r *Router) registerListenOnlyPeer(rid string, uid string, jsep string, limit int, callback string) (string, *webrtc.SessionDescription, error) {
```

Comment:
It's better to have a new param for the publish method, to allow create a peer without offer.

Reply:
That is we don't need a new method registerListenOnlyPeer

Reply:
Sorry I lost the context that we have discussed this in the issue before this PR. 

I think at first, we need to add a publish method param similar to the `limit`, which means whether allows others to listen only without publish.

Then in the subscribe method, if no peer found and publishers allow listen only, we will create a new peer for the subscriber.

Reply:
I see what you mean - this seems more efficient. I'll make the updates and push the changes in the next day or so.